src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+                    return (subgroups_index, commands_index)
+
+                self.assertGreater(len(commands), commands_index,
+                                   msg='None or missing expected commands provided in test for ' + command_input) #pylint: disable=line-too-long
+                if first_word_of_line == commands[commands_index]:
+                    # make sure there is descriptive text in this line
+                    first_index = line.find(': ')
+                    self.assertGreater(
+                        len(line),
+                        len(first_word_of_line),
+                        msg='Validating help text failed in "Commands" on line: ' + line)
+
+                    commands_index += 1
+
+            elif section == 'Subgroups':
+
+                # If we've reached the end of the commands tuple, then skip
+                if len(subgroups) == subgroups_index:
+                    return (subgroups_index, commands_index)
+
+                self.assertGreater(len(subgroups), subgroups_index,
+                                   msg='None or missing expected subgroups provided in test for ' + command_input) #pylint: disable=line-too-long
+                if first_word_of_line == subgroups[subgroups_index]:
+                    # make sure there is descriptive text in this line
+                    first_index = line.find(': ')
+                    self.assertGreater(
+                        len(line),
+                        len(first_word_of_line),
+                        msg='Validating help text failed in "Subgroups" on line: ' + line)
+
+                    subgroups_index += 1
+
+            return (subgroups_index, commands_index)
+
+        else:
+            self.fail('Section name {0} is not supported'.format(section))
+
+    @classmethod
+    def _validate_output_read_section_name(cls, line):
+        """ Read a given line and validate it for correctness based on the given section.
+        Parameter line (string) should be unprocessed. For example, the line should not be
+        stripped of starting or trailing white spaces.
+
+        Returns the section name if the given line designates the beginning of a new section.
+        Returns None if the line does not. """
+
+        if (line.strip() and (line[0] != ' ')):
+            # Use these lines to set the 'section' variable and move on to the next line
+            line = line.strip().rstrip(':')
+            if line == 'Commands':
+                return 'Commands'
+            elif (line == 'Arguments') or (line == 'Global Arguments'):
+                return 'Arguments'
+            elif line == 'Group':
+                return 'Group'
+            elif line == 'Subgroups':
+                return 'Subgroups'
+            elif line == 'Command':
+                return 'Command'
+        return None
+
+    def validate_output(self, command_input, subgroups=(), commands=()): #pylint: disable=too-many-locals
```

Comment:
Use `None` not `()`

Reply:
I thought this was only an issue for mutable default values? 

I added this because later in the code I check `len(commands)` and `len(subgroups)` and I wanted to avoid a check for None first.